### PR TITLE
allow site-replication checks to work on non-distributed setups

### LINKED ISSUE #17523: Site replication broken in RELEASE.2023-06-23T20-26-00Z
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

## Expected Behavior
Site replication should work

## Current Behavior
Site replication throws error 'XMinioAdminRemoteIdenticalToSource'
The remote target cannot be identical to source (remote target endpoint http://192.168.0.10:9000 is self referential)"
*This happens when using two completely physically separate machines

## Possible Solution
Rolled back to RELEASE.2023-06-19T19-52-50Z and it works as intended

## Steps to Reproduce (for bugs)
Download fresh copy of RELEASE.2023-06-23T20-26-00Z and enable site replication between two sites

Buckets are replicated but files are not, error is XMinioAdminRemoteIdenticalToSource

## Context

I upgraded to latest and needed to enable replication but does not work on latest, spent hours double checking configuration but it turns out its just a version issue in the latest

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
allow site-replication checks to work on non-distributed setups

## Motivation and Context
fixes #17523

## How to test this PR?
Well as per #17523 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression yes perhaps introduced by a combination of #17467 and #17474
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/server-main.go [hunk 1]
```diff
@@ -641,6 +632,15 @@ func serverMain(ctx *cli.Context) {
 	xhttp.SetDeploymentID(globalDeploymentID)
 	xhttp.SetMinIOVersion(Version)
 
+	for _, n := range globalNodes {
+		nodeName := n.Host
+		if n.IsLocal {
+			nodeName = globalLocalNodeName
+		}
+		nodeNameSum := sha256.Sum256([]byte(nodeName + globalDeploymentID))
```
Comment (vadmeste) [inline cmd/server-main.go:640]: yes, globalDeploymentID is needed

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Comment by ShayneLee8** [comment]
very good,it has solved my problem

**Comment by sunsjgit** [comment]
Thank you so much,  After upgrading the version, this issue was resolved ！  very good ！

**Pull request merged**